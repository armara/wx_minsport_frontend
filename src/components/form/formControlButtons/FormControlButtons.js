import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/button/Button'

import 'components/form/formControlButtons/FormControlButtons.scss'

const FormControlButtons = ({
  cancelText,
  successText,
  successType,
  onCancelClick,
  onSuccessClick,
}) => (
  <div className="form-control-buttons">
    <Button className="control-buttons_cancel" onClick={onCancelClick}>
      {cancelText}
    </Button>
    <Button
      className="is-success control-button-save"
      type={successType}
      onClick={onSuccessClick}
    >
      {successText}
    </Button>
  </div>
)

FormControlButtons.defaultProps = {
  cancelText: 'Отмена',
  successText: 'Сохранить',
  successType: 'submit',
  onCancelClick: () => {},
  onSuccessClick: () => {},
}

FormControlButtons.propTypes = {
  cancelText: PropTypes.string,
  successText: PropTypes.string,
  successType: PropTypes.string,
  onCancelClick: PropTypes.func,
  onSuccessClick: PropTypes.func,
}

export default FormControlButtons
