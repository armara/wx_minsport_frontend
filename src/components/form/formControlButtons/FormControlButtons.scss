.form-control-buttons {
  display: flex;
  justify-content: space-between;
  margin-top: 20px;

  > button.base-button {
    width: auto;
  }
}