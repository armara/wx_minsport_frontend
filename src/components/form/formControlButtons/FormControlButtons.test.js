import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import FormControlButtons from 'components/form/formControlButtons/FormControlButtons'

configure({ adapter: new Adapter() })

describe('<FormControlButtons />', () => {
  it('renders correctly default props', () => {
    const tree = renderer.create(<FormControlButtons />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
