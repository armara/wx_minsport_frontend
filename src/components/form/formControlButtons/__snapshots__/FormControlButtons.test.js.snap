// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControlButtons /> renders correctly default props 1`] = `
<div
  className="form-control-buttons"
>
  <button
    className="control-buttons_cancel button base-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    title=""
    type="button"
  >
    
    Отмена
  </button>
  <button
    className="is-success control-button-save button base-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    title=""
    type="submit"
  >
    
    Сохранить
  </button>
</div>
`;
